subprojects/core-api/src/main/java/org/gradle/api/artifacts/transform/ArtifactTransformOutputs.java

```diff
@@ -48,12 +48,12 @@
      *
      * @param output the output
      */
-    void registerOutputFile(File output);
+    void registerOutput(File output);
 
     /**
      * Registers the provided workspace as the output of the transform.
      *
      * @return location of the workspace directory
      */
-    File registerWorkspaceAsOutputDirectory();
+    File registerWorkspaceAsOutput();
```

Comment:
I think a little explanation about what registering the whole workspace as output means would be nice.